### ISSUE TITLE
Problems pane ignores "Hide Excluded Files" option

### ISSUE THREAD

**Issue by Sergej-Popov** [microsoft/vscode#103631]
- VSCode Version: 1.47.2
- OS Version: Windows 10 (2004)

Steps to Reproduce:

1. Create a dotnet project
2. Add `**\obj` & `**\bin` to `settings.json` `files.exclude`
3. Open a problems pane and check "Hide Excluded Files"
4. Build and open problems

Expected: problems pane not to show problems in generated code in `**\obj` folder.
Actual: all problems are still visible under problems pane but as expected not visible in the explorer.

![image](https://user-images.githubusercontent.com/1652990/88928233-7b98c100-d270-11ea-9597-02dc897a215f.png)

Does this issue occur when all extensions are disabled?: No, the "problems" are generated by an extension. So cannot reproduce this way.

Related issue: https://github.com/microsoft/vscode/issues/22289

**Comment by gjsjohnmurray** [comment]
AFAIK the feature in Problems to exclude items uses its own filter, which you must enter in the field to the left of the filter button. I don't think it pays any attention to `files.exclude`

**Comment by Sergej-Popov** [comment]
This comment suggests this `files.exclude` suppose to work: https://github.com/microsoft/vscode/issues/22289#issuecomment-380611244

While I can use the on the left but it is separate from "Hide Excluded Files" option. They don't interact.

**Comment by andrew-maust** [comment]
Hi I started to investigate this bug and I believe that what @Sergej-Popov is actually the expected behavior due to the fact that the editor is visible. Unless the problems explorer is displaying problems from more excluded files besides the one(s) open. https://github.com/microsoft/vscode/blob/42a9a9998e72756ef598433bb5699545c99fcd69/src/vs/workbench/contrib/files/browser/views/explorerViewer.ts#L509-L516

**Comment by Sergej-Popov** [comment]
Clarification with regards to https://github.com/microsoft/vscode/blob/42a9a9998e72756ef598433bb5699545c99fcd69/src/vs/workbench/contrib/files/browser/views/explorerViewer.ts#L509-L513

Problems are always visible not only when the file is open in the editor. I opened one in the screenshot just to show the error, but others that aren't open in the editor are still visible in the Problems pane too.

**Comment by andrew-maust** [comment]
Oh alright, I realized I misunderstood so that is why I deleted my comment. I apologize for any confusion.

**Comment by andrew-maust** [comment]
Hi upon further investigation it appears that this feature uses a different list. I believe that it goes off of this list found in settings
<img width="981" alt="Screen Shot 2020-08-07 at 10 13 42 AM" src="https://user-images.githubusercontent.com/69081050/89654582-dd6fb100-d896-11ea-8623-3ee0a3d1468e.png">

**Comment by LC43** [comment]
this also happens with php language server. 
is there a 
```json
"php.problems.exclude" : {
        "node_modules/**" : true,
    },
```
of some sort, like [devsense extension](https://docs.devsense.com/en/vscode/problems) has?

I have to disable `"php.validate.executablePath": "",` altogether because `"php.validate.enable": false,` doesn't stop vscode to fill Problems pane.

**Comment by gjsjohnmurray** [comment]
@Sergej-Popov I wasn't able to reproduce this issue when I tested just now on 1.52.1. Problems reported on a file whose name matched a glob in `files.exclude` disappeared from the panel when I selected the "Hide Excluded Files" filter. They reappeared when I deselected that filter option, or when I removed or disabled the `files.exclude` entry.

If the problem still exists for you please provide simple repro steps.

**Comment by sandy081** [comment]
@gjsjohnmurray Very much appreciated for validating the issue again. 

@Sergej-Popov Please provide exact steps to repro.

**Comment by Sergej-Popov** [comment]
Hey @sandy081 @gjsjohnmurray I did a repro on clean machine and I know exactly what the issue is now.

It has to with with how glob pattern is applied in filter few.
If you apply it on folder like this - it does not work for problems pane, but works in explorer view.
```json
{
    "files.exclude": {
        "**/test": true
    }
}
```

If you apply it on all files inside the folder - it works for problems pane and explorer view.
```json
{
    "files.exclude": {
        "**/test/**": true
    }
}
```

I was able to reproduce this behaviour in typescript solution too, so dotnet was a red herring.
![vscode-103631-repro](https://user-images.githubusercontent.com/1652990/106360149-c2926c80-630e-11eb-9755-a0e218e5ea89.gif)
